master/internal/api_workspace.go

```diff
@@ -551,11 +558,16 @@ func (a *apiServer) DeleteWorkspace(
 		return nil, err
 	}
 
+	modelsExist, _ := a.workspaceHasModels(ctx, req.Id)
+	if modelsExist {
+		return nil, fmt.Errorf("workspace (%d) contains models; move or delete models first", req.Id)
+	}
+
 	holder := &workspacev1.Workspace{}
-	err = a.m.db.QueryProto("deletable_workspace", holder, req.Id)
+	// TODO(kristine): update workspace state in transaction with template delete
+	_ = a.m.db.QueryProto("deletable_workspace", holder, req.Id)
 	if holder.Id == 0 {
-		return nil, errors.Wrapf(err, "workspace (%d) does not exist or not deletable by this user",
-			req.Id)
+		return nil, fmt.Errorf("workspace (%d) does not exist or not deletable by this user", req.Id)
```

Comment:
this line is not the same as above we need to specify the error we are wrapping

so for example
```
		return nil, fmt.Errorf("workspace (%d) does not exist or not deletable by this user: %w", req.Id, err)
```

Reply:
I did something a little different; I used a separate check for `err != nil` and `holder.Id == 0`. I think it will allow for more sensible error messages; **what do you think?**

Reply:
Oh - `holder.Id==0` was used as a substitute for checking for an error?